src/analyzer/analyzersilence.cpp

```diff
@@ -117,29 +102,37 @@ void AnalyzerSilence::storeResults(TrackPointer tio) {
     double introStart = mixxx::kAnalysisChannels * m_iSignalStart;
     double outroEnd = mixxx::kAnalysisChannels * m_iSignalEnd;
 
-    if (shouldUpdateMainCue(tio->getCuePoint())) {
-        tio->setCuePoint(CuePosition(introStart, Cue::AUTOMATIC));
+    CuePointer pAudibleSound = pTrack->findCueByType(Cue::Type::AudibleSound);
+    if (pAudibleSound == nullptr || pAudibleSound->getLength() <= 0) {
```

Comment:
I think we can get rid of that condition. Its only purpose was for users who tested an earlier version of this branch that had two separate FirstSound/LastSound cues. Otherwise if the AudibleSound cue exists, it should never have a length <= 0.